src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -168,5 +168,21 @@ public int MaxModelValidationErrors
         /// <see langword="false"/> by default.
         /// </summary>
         public bool AllowBindingUndefinedValueToEnumType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the flag to determine if model binding should convert all exceptions(including ones not related to bad input)
+        /// that occur during deserialization in <see cref="IInputFormatter"/>s into model state errors.
+        /// This flag applies only to custom <see cref="IInputFormatter"/>s. For modifying the behavior of in-built
+        /// <see cref="IInputFormatter"/>s, see <see cref="ConvertAllExceptionsOfInbuiltInputFormattersToModelStateErrors"/>.
+        /// Defaults is <see langword="true"/>.
+        /// </summary>
+        public bool ConvertAllExceptionsToModelStateErrors { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets the flag to determine if model binding should convert all exceptions(including ones not related to bad input)
+        /// that occur during deserialization in <see cref="IInputFormatter"/>s into model state errors.
+        /// This flag applies only to in-built <see cref="IInputFormatter"/>s. Defaults is <see langword="false"/>.
+        /// </summary>
+        public bool ConvertAllExceptionsOfInbuiltInputFormattersToModelStateErrors { get; set; } = false;
```

Comment:
What's the purpose of this for built in formatters? I don't think we should have this as an option